src/Magnum/Octree/Octree.h

```diff
@@ -0,0 +1,274 @@
+
+            if(Math::Geometry::Intersection::pointPlaneHalfSpace<Float>(corner, plane) >= 0) {
+                ++corners;
+            }
+        }
+
+        if(corners == 0) {
+            /* all corners are outside this plane */
+            return false;
+        }
+
+        if(corners == 8) {
+            ++planes;
+        }
+    }
+
+    if(planes == 6) {
+        return true;
+    }
+
+    /* potentially check corners here! */
+    return true;
+}
+
+}}}}
+
+namespace Magnum { namespace Octree {
+
+
+enum class OctantStatus: Byte {
+    NoCorners = 0,   /**< No corners of the octant are inside the testing volume */
+    SomeCorners = 1, /**< Neither all nor no corners of the octant are inside the testing volume */
+    AllCorners = 2,  /**< All corners of the octant are inside the testing volume */
+};
+
+/**
+ * @brief Octree @see https://en.wikipedia.org/wiki/Octree
+ *
+ * An octree is a cubic structure, which is subdivided in up to 8 smaller cubes,
+ * called octants. Each octant can be seen as an octree itself. Data is stored
+ * in the leaf octants (octants without children).
+ */
+template<typename T>
+class Octree {
+    public:
+
+        /** @brief Default constructor */
+        Octree() :
+            _center{},
+            _radius(),
+            _depth(),
+            _isLeafNode()
+        {
+            for(int i = 0; i < 8; ++i) {
+                _children[i] = nullptr;
+            }
+        }
+
+        /** @brief Default destructor */
+        ~Octree()=default;
+
+        /** @brief Get the center of the octree */
+        Vector3 center() const {
+            return _center;
+        }
+
+        /** @brief Get the radius of the octree. */
+        Float radius() const {
+            return _radius;
+        }
+
+        /** @brief Get the depth of the octree */
+        Int depth() const {
+            return _depth;
+        }
+
+        /** @brief Get the bounding boxes in this octree node */
+        const std::vector<Range3D>& entries() const {
+            return _entries;
+        }
+
+        /** @brief Get the data contained in this octree node */
+        const std::vector<T>& data() const {
+            return _data;
+        }
+
+        /** @brief Get the i-th child of this octree */
+        Octree<T>* child(int i) const {
+            return _children[i];
+        }
+
+        /**
+         * @brief Check if the octree is a leaf node.
+         *
+         * An octree is a leaf node if it has no children.
+         */
+        bool isLeafNode() const {
+            return _isLeafNode;
+        }
+
+        /**
+         * @brief Set if octree is a leaf node.
+         *
+         * An octree is a leaf node if it has no children.
+         */
+        void setLeafNode(bool isLeafNode) {
+            _isLeafNode = isLeafNode;
+        }
+
+        /** @brief Whether this node is empty */
+        bool empty() const {
+            return _data.size() == 0;
+        }

```

Comment:
Can you put the `return` statements in the above short functions into one line with function declaration to compress the code vertically a bit?